File: gachisa-backend/src/main/java/com/gachisa/category/dto/CategoryResponse.java
package com.gachisa.category.dto;

import com.gachisa.category.entity.Category;
import java.util.List;
import java.util.stream.Collectors;

public record CategoryResponse(
    Long id,
    String name,
    Long parentId,
    List<CategoryResponse> children
) {
    public static CategoryResponse from(Category category) {
        Long parentId = category.getParent() != null ? category.getParent().getId() : null;
        List<CategoryResponse> children = category.getChildren().stream()
            .map(CategoryResponse::from)
            .collect(Collectors.toList());
        return new CategoryResponse(category.getId(), category.getName(), parentId, children);
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/category/dto/CategoryResponse.kt
package com.gachisa.category.dto

import com.gachisa.category.entity.Category

data class CategoryResponse(
    val id: Long?,
    val name: String,
    val parentId: Long?,
    val children: List<CategoryResponse>,
) {
    companion object {
        fun from(category: Category): CategoryResponse {
            return CategoryResponse(
                id = category.id,
                name = category.name,
                parentId = category.parent?.id,
                children = category.children.map { from(it) },
            )
        }
    }
}
